tests/rptest/tests/isolated_decommissioned_node_test.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2022 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from rptest.services.cluster import cluster
+from rptest.services.admin import Admin
+from ducktape.utils.util import wait_until
+from rptest.clients.types import TopicSpec
+from rptest.tests.prealloc_nodes import PreallocNodesTest
+from rptest.util import firewall_blocked
+from confluent_kafka import (Producer, KafkaException)
+
+import time
+
+
+def on_delivery(err, msg):
+    if err is not None:
+        raise KafkaException(err)
+
+
+class IsolatedDecommissionedNodeTest(PreallocNodesTest):
+    topics = (TopicSpec(partition_count=1, replication_factor=3), )
+
+    def __init__(self, test_context):
+        super(IsolatedDecommissionedNodeTest,
+              self).__init__(test_context=test_context,
+                             num_brokers=3,
+                             node_prealloc_count=3)
+
+        self.internal_port = 33145
+        self.admin = Admin(self.redpanda)
+
+    @cluster(num_nodes=3)
+    def simple_produce_test(self):
+        topic = self.topics[0]
+        with firewall_blocked([self.redpanda.nodes[0]], self.internal_port):
+
+            def is_node_isolated():
+                return self.admin.is_node_isolated(self.redpanda.nodes[0])
+
+            wait_until(is_node_isolated, timeout_sec=20, backoff_sec=1)
+
+            producer = Producer({
+                "bootstrap.servers":
+                self.redpanda.broker_address(self.redpanda.nodes[0]),
+            })
+            producer.produce(str(topic),
```

Comment:
Some comments here would help to explain the test's expectations: on first read it's bit surprising that our test for a node being isolated is to try and produce to it and succeed :-)

I'm guessing the intended behavior is that this client talks to the isolated node, gets the metadata response with other nodes only, and then proceeds to talk to a different node and succeed.